src/components/DragAndDrop/index.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+
+import variables from '../../styles/variables';
+import DragAndDropPropTypes from './dragAndDropPropTypes';
+
+const COPY_DROP_EFFECT = 'copy';
+const NONE_DROP_EFFECT = 'none';
+
+const propTypes = {
+    ...DragAndDropPropTypes,
+
+    onDragOver: DragAndDropPropTypes.onDragOver,
+
+    /** Rendered child component */
+    children: PropTypes.node.isRequired,
+};
+
+const defaultProps = {
+    onDragOver: () => {},
+};
+
+export default class DragAndDrop extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.dropZone = document.getElementById(props.dropZoneId);
+        this.dropZoneRect = this.calculateDropZoneClientReact();
+
+        this.dragOverHandler = _.throttle(this.dragOverHandler.bind(this), 100);
+        this.dragNDropWindowResizeListener = _.throttle(this.dragNDropWindowResizeListener.bind(this), 100);
+        this.dropZoneDragHandler = this.dropZoneDragHandler.bind(this);
+        this.dropZoneDragListener = this.dropZoneDragListener.bind(this);
+
+        /*
+        Last detected drag state on the dropzone -> we start with dragleave since user is not dragging initially.
+        This state is updated when drop zone is left/entered entirely(not taking the children in the account) or entire window is left
+        */
+        this.dropZoneDragState = 'dragleave';
+    }
+
+    componentDidMount() {
+        document.addEventListener('dragover', this.dropZoneDragListener);
+        document.addEventListener('dragenter', this.dropZoneDragListener);
+        document.addEventListener('dragleave', this.dropZoneDragListener);
+        document.addEventListener('drop', this.dropZoneDragListener);
+        window.addEventListener('resize', this.dragNDropWindowResizeListener);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('dragover', this.dropZoneDragListener);
+        document.removeEventListener('dragenter', this.dropZoneDragListener);
+        document.removeEventListener('dragleave', this.dropZoneDragListener);
+        document.removeEventListener('drop', this.dropZoneDragListener);
+        window.removeEventListener('resize', this.dragNDropWindowResizeListener);
+    }
+
+    /**
+     * @param {Object} event native Event
+     */
+    dragOverHandler(event) {
+        this.props.onDragOver(event);
+    }
+
+    dragNDropWindowResizeListener() {
+        // Update bounding client rect on window resize
+        this.dropZoneRect = this.calculateDropZoneClientReact();
+    }
+
+    calculateDropZoneClientReact() {
+        const boundingClientRect = this.dropZone.getBoundingClientRect();
```

Comment:
The error was coming as this line.

Reply:
Yeah, because there was no drop zone ancestor in emoji compose modal. Should be all fine now

Reply:
Can you share the commit link here? What was the fix?